components/helm-broker/internal/broker/catalog_test.go

```diff
@@ -69,60 +76,40 @@ func TestGetCatalogOnConversionError(t *testing.T) {
 func TestBundleConversion(t *testing.T) {
 	// GIVEN
 	tc := newCatalogTC()
-	fixBundle := tc.fixBundle()
+	goldenPath := filepath.Join("testdata", t.Name()+".golden.json")
+
 	conv := broker.NewConverter()
+
 	// WHEN
-	s, err := conv.Convert(fixBundle)
+	convertedSvc, err := conv.Convert(tc.fixBundle())
+
 	// THEN
-	assert.NoError(t, err)
-	assert.Equal(t, "bundleID", s.ID)
-	assert.Equal(t, "bundleName", s.Name)
-	assert.Equal(t, "bundleDescription", s.Description)
-	assert.True(t, s.Bindable)
-	assert.Equal(t, map[string]interface{}{
-		"displayName":         fixBundle.Metadata.DisplayName,
-		"providerDisplayName": fixBundle.Metadata.ProviderDisplayName,
-		"longDescription":     fixBundle.Metadata.LongDescription,
-		"documentationUrl":    fixBundle.Metadata.DocumentationURL,
-		"supportUrl":          fixBundle.Metadata.SupportURL,
-		"imageUrl":            fixBundle.Metadata.ImageURL,
-	}, s.Metadata)
-
-	require.Len(t, s.Plans, 2)
-
-	var p1, p2 osb.Plan
-	if s.Plans[0].ID == "planID1" {
-		p1 = s.Plans[0]
-		p2 = s.Plans[1]
-	} else {
-		p2 = s.Plans[0]
-		p1 = s.Plans[1]
-	}
+	require.NoError(t, err)
+
+	normalizedGotSvc := tc.marshal(t, convertedSvc)
+
+	updateGoldenFileIfRequested(t, goldenPath, normalizedGotSvc)
+
+	exp := tc.fixtureMarshaledOsbService(t, goldenPath)
+	assert.JSONEq(t, exp, string(normalizedGotSvc))
```

Comment:
I tried to assert on Go obj 
using sth like that:
```go
func fixtureOsbService(t *testing.T, testdataBasePath string) osb.Service {
	data, err := ioutil.ReadFile(testdataBasePath)
	require.NoError(t, err, "failed reading .golden")

	out := osb.Service{}
	require.NoError(t, json.Unmarshal(data, &out))
	return out
}
```

Unfortunately kinds are different, e.g. `map[string]string` vs `Labels`
where `Labels` is  `type Labels map[string]string`. Then in the assertion, it cannot be compared easily. Because of that, I decided to normalize it to the JSON form.